### PR TITLE
[ACCELERATE-503]Consuming API

### PR DESCRIPTION
**Pull request by carloscasanovad**
This PR is about changing all the mock data to data from a endpoint and widgets/unit tests.

### JIRA Tracking

- [ACCELERATE-503](https://warren.atlassian.net/browse/ACCELERATE-503?atlOrigin=eyJpIjoiZGM2MWE4OTEwMjkyNDRjZGI4ZTllMjNjZTdkYjgzM2MiLCJwIjoiaiJ9)

### Description

In this PR it was added the whole clean architecture in order to use the endpoints given on the card. The application UI didn't change a bit, since the card was about **consuming api**. Besides that, it was added some widgets and units tests.

ps. It is also important to mention that, the  **"SEARCH ONE"** endpoint was implemented but it wasn't used because it was no need.

### Is this a breaking change

- [ ] Yes
- [x] No

### Screenshots or GIFs

image
image

### CODE REVIEW

File: coverage/lcov.info [hunk 1]
```diff
@@ -1,28 +1,375 @@
-SF:lib/main.dart
-DA:3,0
-DA:4,0
-DA:8,2
+SF:lib\shared\api\movements_base_endpoint.dart
```
Comment (CamileBarros) [inline coverage/lcov.info:1]: isso aqui pode ficar no .gitignore

File: pubspec.lock [hunk 2]
```diff
@@ -155,6 +155,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "3.1.1"
```
Comment (CamileBarros) [inline pubspec.lock:157]: esse arquivo tbm pode ir pro .gitignore

**Review by CamileBarros — APPROVED** [review]
Prestar atenção no número de pastas que são editadas por card. Se algo tiver que ser refatorado, e não for cruacial para atual tarefa que você está desempenhando, você pode abrir uma branch só para isso depois..

File: coverage/lcov.info (hunk 1 above)
Comment (carloscasanovad) [reply coverage/lcov.info:1]: Ok

File: pubspec.lock (hunk 2 above)
Comment (carloscasanovad) [reply pubspec.lock:157]: Feito

**Pull request merged**